Marlin/src/HAL/STM32/Sd2Card_sdio_stm32duino.cpp

```diff
@@ -88,18 +88,37 @@
 
     MKS Robin board seems to have stable SDIO with BusWide 1bit and ClockDiv 8 i.e. 4.8MHz SDIO clock frequency
     Additional testing is required as there are clearly some 4bit initialization problems
-
-    Add -DTRANSFER_CLOCK_DIV=8 to build parameters to improve SDIO stability
   */
 
-  #ifndef TRANSFER_CLOCK_DIV
-    #define TRANSFER_CLOCK_DIV (uint8_t(SDIO_INIT_CLK_DIV) / 40)
-  #endif
-
   #ifndef USBD_OK
     #define USBD_OK 0
   #endif
 
+  // Target Clock, configurable. Default is 18MHz, from STM32F1
+  #ifndef SDIO_CLOCK
+    #define SDIO_CLOCK                         18000000       /* 18 MHz */
+  #endif
+
+  // SDIO retries, configurable. Default is 3, from STM32F1
+  #ifndef SDIO_READ_RETRIES
+    #define SDIO_READ_RETRIES                  3
+  #endif
+
+  // SDIO Max Clock (naming from STM Manual, don't change)
+  #define SDIOCLK 48000000
+
+  static uint32_t sdio_set_clock(uint32_t clk) {
```

Comment:
This name could be better. It is not easy to tell this returns a clock divider.

Reply:
Perhaps `sdio_calc_clock_divider`?

Reply:
I agree that the name is bad. I thought in changing it too.. But in the end I kept it, because si the same name as maple.

Reply:
I don't care about Maple.

Reply:
By which I mean that we should not allow badly named Maple code to force bad names on us here.